esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -10,6 +11,62 @@
 from esmf_regrid.experimental.unstructured_regrid import MeshInfo
 
 
+def _map_complete_blocks(src, func, dims, out_sizes):
+    """
+    Apply a function to complete blocks.
+
+    Complete means that the data is not chunked along the chosen dimensions.
+
+    Parameters
+    ----------
+    src : cube
+        Source :class:`~iris.cube.Cube` that function is applied to.
+    func : function
+        Function to apply.
+    dims : tuple of int
+        Dimensions that cannot be chunked.
+    out_sizes : tuple of int
+        Output size of dimensions that cannot be chunked.
+
+    Returns
+    -------
+    cube
+        A new iris.cube.Cube instance.
+
+    """
+    if not src.has_lazy_data():
+        return func(src.data)
+
+    data = src.lazy_data()
+
+    # Ensure dims are not chunked
+    in_chunks = list(data.chunks)
+    for dim in dims:
+        in_chunks[dim] = src.shape[dim]
+    data = data.rechunk(in_chunks)
+
+    # Determine output chunks
+    out_chunks = list(data.chunks)
+    num_dims = len(dims)
+    num_out = len(out_sizes)
+    sorted_dims = list(dims)
+    sorted_dims.sort()
+    dropped_dims = []
+    for i in range(max(num_dims, num_out)):
```

Comment:
This loop needs a comment - overview of what it is checking for in the multi-if.